Use formik nested objects

souad988 opened the pull request:
## Summary

Description of PR here...
Changed form state management to be fully handled by formik .
Add deserialize function to refactor activity object and fill the form for edit.
Add video upload validation for video file or url and disable video inputs after one is used.
Add save activity functionality.

souad988 committed:
Create activity reducer with initial state for all activities

souad988 committed:
Create activity actions file with setActivities function

souad988 committed:
Update configureStore to include activityReducer

souad988 committed:
Created a generic activities objects to use meanwhile creating backend for activities

souad988 committed:
Create Activities component and add button in header to render it

souad988 committed:
Connect activities component to activities store

souad988 committed:
Create action types constants

souad988 committed:
Remove unused files

souad988 committed:
Create activities video carousel

souad988 committed:
Create activities video carousel

souad988 committed:
Create activity card

souad988 committed:
Add projects count icon and activity tags to the card

souad988 committed:
Add creator to activity card

souad988 committed:
Create creator component to handle multipe activity creator case and display them in activity card

souad988 committed:
Show full creator image on hover

souad988 committed:
Ajust activity cards and handle display of more than 3 tags

souad988 committed:
Create activity creation form container

souad988 committed:
Create form steps components and form progressBar

souad988 committed:
Add translation for BROWS ACTIVITIES Button

souad988 committed:
Merge branch 'activities_store' into create_activity_form

souad988 committed:
Create reusable dynamic progress bar

souad988 committed:
Create activity creation step1 form

souad988 committed:
Create create activity step 2 form inputs

souad988 committed:
Create separate component for react quill text fields

souad988 committed:
Create AddMore and MaterialsUsed component

souad988 committed:
Add handleOnChange an on blur for materials used inputs

souad988 committed:
Create formLabel and UploadFile components

souad988 committed:
Add validation to step1 fields

souad988 committed:
Create step3 input blocks

souad988 committed:
Add form steps verification

souad988 committed:
Add defaultValue to input and inputText to display values entered when checking back previous steps

souad988 committed:
Add handleImageChange to handle all fileUpload instances and fix form responsiveness issues

souad988 committed:
Add validation to fields accepting images and Create upload image functionality

souad988 committed:
Add at least one material required validation for materialsUsed array field

souad988 committed:
Fix inspiringExemplesImages append files issue

souad988 committed:
Try to fix appending multiple inspiringExemplesImages images to newActivity state by adding keys to uploadFiles component instances and add promise to set state

souad988 committed:
Create separate states for step verifiedStep and newActivity and use name instead of label in IputText and send only required props as parameter

souad988 committed:
Update InputText and Input components and according script functions to accept only needed props

souad988 committed:
Update materialsUsed and all other fields using Input and InputText components and make them update form state correctly

souad988 committed:
Update UploadFiles and uploadFile scripts to accept the new states and handle multiple inputs feeding same field in the state issue

souad988 committed:
Upload images to local and setNewActivityObject with urls

souad988 committed:
Handle video change on uploadFile

souad988 committed:
Fix inspiringExemplesImages files_to_upload issue to upload correctly

souad988 committed:
updateState with formik file values on form submit

souad988 committed:
Add hided submitButton with ref and click on setState from refactorNewActivityObject to upload files after state is updated

souad988 committed:
Create activities app backend

souad988 committed:
Create activities app

souad988 committed:
Add materials_udes_image field to activity model

souad988 committed:
Add category and add order to making steps

souad988 committed:
Add inspiring artists

souad988 committed:
started updating handleFileFieldChange

souad988 committed:
Add inspired_from_activity field to Project model and run migrations

souad988 committed:
Create UploadMedia and FileField classes to handle file input changes

souad988 committed:
Add inputLabel to display label in inputs not having FormLabel and Add artist full name input to artist section

souad988 committed:
Move upload media to uploadFileScript and Change it to return promise and create appendFile to field for file field accepting multiple files

souad988 committed:
Create activity serializers and ListAPIView

souad988 committed:
Merge create activity form with create activity backend app for tests

souad988 committed:
Create activity serializers create method and combine fields data for nested objects in createActivityScripts

souad988 committed:
Add creator to created activity

souad988 committed:
Fixing input value undefined

souad988 committed:
Fix styled-components package import issue

souad988 committed:
Merge branch 'activity_card_creator' into merge_form_and_api_for_tests

souad988 committed:
Refactor activity object returned from backend to fit form fields for edit and Add redirection and routes for activity edit

souad988 committed:
Fix Input and InputText display value from state issue

souad988 committed:
Create map of field names and activity object attribute to help deserialize and display data in the form for edit

souad988 committed:
Successfully Update activity and all related fields

souad988 committed:
Fix delete activity issue and add loading to activities veiw

souad988 committed:
Remove commented code from activities view

souad988 committed:
Fix refresh activities list after delete and update issue by using useSelector hook instead of reading reducer state from props

souad988 committed:
Fix Display InputText errors issue

souad988 committed:
Remove prev selected files if new files have errors and display error

souad988 committed:
Add preview image feature to UploadFile component

souad988 committed:
Add preview image to UploadFile component

souad988 committed:
Preview image delete works, add validation for activity images on update when deleting all urls

souad988 committed:
Fix activity without unrequired fields creation issue by removing field keys from api object instead of sending empty objects

souad988 committed:
Changed all form state to be handled by formik including array of objects fields and updated create activity

souad988 committed:
Update step verification to verify step when required field provided with no error and let user see msgs

souad988 committed:
Add function to fill form from activity object and call update activity

souad988 committed:
update and create working with no issues after refactor

souad988 committed:
Use formik field Array to display making steps and inspiring examples and fix all update issues now creation and update working correctly

souad988 committed:
Add  empty object validation and file compression

souad988 committed:
Handle video validation Upload and preview by UploadFile component

souad988 committed:
Add save activity functionality and first section of activity details view

souad988 committed:
Add verification on clicking next button

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      "title": {
+        "label": "Name your Activity",
+        "errors": {
+          "max": "your title shouldn't be more than 100 characters",
```
Change text to "Less than 100 characters"

srish replied:
Re-thinking, I think we should reduce the limit to 50 chars

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+    },
+    "inputs": {
+      "title": {
+        "label": "Name your Activity",
```
Change placeholder text to "Provide a title for your activity"

srish replied:
Change text to "Title"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+        }
+      },
+      "motivation": {
+        "label": "motivation",
```
Make the "m" capital in the label so it becomes "Motivation"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      },
+      "motivation": {
+        "label": "motivation",
+        "helperText": "Add motivation text",
```
Remove helper text

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+        "helperText": "Add motivation text",
+        "placeholder": "Add motivation text",
+        "errors": {
+          "max": "your motivation shouldn't be more than 10,000 characters",
```
Change this text to "Less than 10000 characters"

srish replied:
Re-thinking, I think we should reduce the limit to 500 chars

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+        }
+      },
+      "learningGoals": {
+        "label": "Add Learning Goals",
```
Remove "Add" from "Add Learning Goals"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+        "helperText": "List some of this activity's learning goals",
+        "placeholder": "List some of this activity's learning goals",
+        "errors": {
+          "max": "your learning goals shouldn't be more than 10,000 characters",
```
Change the text to "Less than 1000 characters" and reduce the limit

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      },
+      "learningGoals": {
+        "label": "Add Learning Goals",
+        "helperText": "List some of this activity's learning goals",
```
Remove helperText

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+        }
+      },
+      "materialsUsed": {
+      "label": "what materials did you use?",
```
Change text to "Materials"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      },
+      "materialsUsed": {
+      "label": "what materials did you use?",
+      "helperText": "List some of the essential materials used to create this activity",
```
Change text "What materials did you use to develop this activity?" Where are we currently displaying this because I don't see it.

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      "materialsUsed": {
+      "label": "what materials did you use?",
+      "helperText": "List some of the essential materials used to create this activity",
+      "encouragingText": "We strongly encourage you to upload an image containing all materials and tools used to create this activity",
```
Change text to "Add an image with all materials used"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      "helperText": "List some of the essential materials used to create this activity",
+      "encouragingText": "We strongly encourage you to upload an image containing all materials and tools used to create this activity",
+      "images": {
+        "label": "Add Images"
```
It is a bit confusing that we are allowing uploading of only one image but here we say "Images" in the label. Maybe we say "Add Image"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+        "label": "Add Images"
+      },
+      "errors": {
+        "max": "your material shouldn't be more than 50 characters",
```
This error is not populating, instead I am seeing `createActivity.inputs.materials_used[0].errors.max`. In any case, let's change the text to "Less than 50 characters"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      },
+      "errors": {
+        "max": "your material shouldn't be more than 50 characters",
+        "required": "this field is required",
```
Let's make t in "this" capital for consistency.

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+        "max": "your material shouldn't be more than 50 characters",
+        "required": "this field is required",
+        "required1": "At least one material is required!",
+        "min": "your material should be at least 2 characters"
```
Let's change text to "Atleast two characters are required"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      "errors": {
+        "max": "your material shouldn't be more than 50 characters",
+        "required": "this field is required",
+        "required1": "At least one material is required!",
```
Let's change text to "Atleast one material is required"; so remove "!" mark for consistency and join "At" and "least".

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      },
+      "facilitationTips": {
+        "label": "Facilitation tips",
+        "helperText": "List some helpful facilitation tips",
```
Remove helper text

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+        "placeholder": "Facilitation tips",
+        "errors": {
+          "max": "your facilitation tips shouldn't be more than 10,000 characters",
+          "required": "This field is required"
```
Change the text to "Less than 1000 characters" and reduce the limit.

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      }
+      },
+      "facilitationTips": {
+        "label": "Facilitation tips",
```
Make "t" in "tips" capital for consistency

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+
+      },
+      "activityImages": {
+        "label": "Add Images of your activity",
```
Change text to "Demo Images"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+        "label": "Add Images of your activity",
+        "errors": {
+          "required1": "you can't create a new activity without images",
+          "required": "This field is required",
```
Can we just use "required" and get rid of "required1"?

srish commented on zubhub_frontend/zubhub/src/views/create_activity/create_activity_step2.jsx:
```diff
@@ -0,0 +1,171 @@
+            classes={classes}
+            common_classes={common_classes}
+            inputOrder={6}
+            fieldLabel={'Add a video if provided'}
```
Change the text here to "Demo Videos". I was wondering about the placement of this label separate from others as well, but not a big deal.

srish commented on zubhub_frontend/zubhub/src/views/create_activity/create_activity_step2.jsx:
```diff
@@ -0,0 +1,171 @@
+          className={common_classes.marginTop1em}
+        >
+          <FormLabel
+            label={'ActivityVideo'}
```
When the video exceeds a limit, error is getting displayed as `createActivity.inputs.video.errors.maxVideoFileSize` on the screen instead of text

srish commented on zubhub_frontend/zubhub/src/views/create_activity/create_activity_step2.jsx:
```diff
@@ -0,0 +1,171 @@
+            <UploadFile
+              name={'video'}
+              fileType={'video/*'}
+              uploadButtonLabel={'Upload Video'}
```
I think we should change text here to "Add Video" because for adding video URLs that makes more sense.

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+        }
+      },
+      "making_steps": {
+        "label": "Creation Steps",
```
Change text to "Making Process"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+          "label": "Add Step Image"
+        },
+        "errors": {
+          "max": "your creation step shouldn't be more than 10,000 characters",
```
Change text to "1000" characters and reduce the limit

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+        "placeholder": "creation step",
+        "addMore": "Add a Step",
+        "image": {
+          "label": "Add Step Image"
```
Change text to "Add Image"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      "making_steps": {
+        "label": "Creation Steps",
+        "helperText": "List steps you followed to create this activity",
+        "placeholder": "creation step",
```
Change the text to "What was your process for developing the activity? What were the various steps involved?"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      },
+      "making_steps": {
+        "label": "Creation Steps",
+        "helperText": "List steps you followed to create this activity",
```
Remove helperText

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      "facilitationTips": {
+        "label": "Facilitation tips",
+        "helperText": "List some helpful facilitation tips",
+        "placeholder": "Facilitation tips",
```
Change text to "How can educators engage learners in such an activity? What questions can they ask a learner about the activity? What kind of examples can they demonstrate?"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      "learningGoals": {
+        "label": "Add Learning Goals",
+        "helperText": "List some of this activity's learning goals",
+        "placeholder": "List some of this activity's learning goals",
```
Change text to "What will someone learn from this activity?"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      "motivation": {
+        "label": "motivation",
+        "helperText": "Add motivation text",
+        "placeholder": "Add motivation text",
```
Change the text to "Share your motivation for developing the activity"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+        },
+        "errors": {
+          "max": "your creation step shouldn't be more than 10,000 characters",
+          "required": "describe creation process in at least one step ",
```
Change text to "Atleast one step is required"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+        "errors": {
+          "max": "your creation step shouldn't be more than 10,000 characters",
+          "required": "describe creation process in at least one step ",
+          "emptyStep": "step can't be empty provide either a description or an image"
```
Change text to "Provide either an image or description for a step"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+        }
+      },
+      "inspiringArtist": {
+        "label": "Inspiring Artist",
```
Perhaps we should change this to "Inspiring Person"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      "inspiringArtist": {
+        "label": "Inspiring Artist",
+        "helperText": "Enter a Bio of your inspiring artist",
+        "placeholder": "Inspiring artist Bio",
```
Share the background of an inspiring artist, inventor, or educator who is the inspiration behind this activity or whose work relates to it

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      },
+      "inspiringArtist": {
+        "label": "Inspiring Artist",
+        "helperText": "Enter a Bio of your inspiring artist",
```
Remove helpertext

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+        "helperText": "Enter a Bio of your inspiring artist",
+        "placeholder": "Inspiring artist Bio",
+        "image": {
+          "label": "Add Artist Image"
```
Change text to "Add Image"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+          "label": "Add Artist Image"
+        },
+        "errors": {
+          "max": "your artist bio shouldn't be more than 10,000 characters",
```
Less than 1000 characters and reduce the limit

srish commented on zubhub_frontend/zubhub/src/views/create_activity/create_activity_step3.jsx:
```diff
@@ -0,0 +1,448 @@
+            )}
+          >
+            <Input
+              label={'ArtistFullName'}
```
Lets change text to "Name"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+        }
+      },
+      "inspiringExemples": {
+        "label": "inspiring Examples",
```
I capital in "inspiring" for consistency

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+        "label": "inspiring Examples",
+        "addMore": "Add Example",
+        "image": {
+          "label": "Add Example Image"
```
Change text to "Add Image"

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+          "required": "This field is required"
+        }
+      },
+      "inspiringExemples": {
```
this variable has spelling mistake..

srish commented on zubhub_frontend/zubhub/src/views/create_activity/create_activity_step3.jsx:
```diff
@@ -0,0 +1,448 @@
+                      <Grid container direction="row" spacing={3}>
+                        <Grid item xs={12} md={4} sm={6}>
+                          <Input
+                            label={`Img-${key + 1} desc`}
```
Let's change this so it appears as "1. Image description"

srish commented on zubhub_frontend/zubhub/src/views/create_activity/create_activity_step3.jsx:
```diff
@@ -0,0 +1,448 @@
+                        </Grid>
+                        <Grid item xs={12} md={3} sm={6}>
+                          <Input
+                            label={`Img-${key + 1} credit`}
```
Let's change this so it appears "1. Image credit"

srish commented on zubhub_frontend/zubhub/src/components/materialsUsed/materialsUsed.jsx:
```diff
@@ -0,0 +1,131 @@
+        <Grid item xs={12} md={6}>
+          {materialsUsedList.map((value, index) => (
+            <Input
+              label={`material-${index + 1}`}
```
Let's change this so it appears as:
1.
2.
3.

we need only number not material word for this one..

souad988 replied:
I got it 👍

srish commented on zubhub_frontend/zubhub/public/locales/en/translation.json:
```diff
@@ -756,5 +758,148 @@
+      "videoURL": "Video URL added"  
+    },
+    "buttons": {
+      "Next": "Next->",
```
Use the same icons we are using for "previous and next page buttons" on the homepage

srish reviewed (changes requested):
@souad988 Most of the changes I've suggested include text changes, reducing the character limit for fields, and a few minor issues here and there. I have not come across any major issue so far, all suggestions have to do with minor improvements so that's a good thing :)

First, about helper text and placeholders. I think we don't need both. Let's remove helper text and only keep the placeholder. Maybe there is one place where we might need to retain helper text (likely materials). Use the text I've suggested for various fields. Once we finalize the changes, I can suggest some changes for the translation file for "Hindi."

Then, I saw that there is quite a lot of stuff that gets printed on the console. Make sure there are no errors, and they don't get printed as well.

Lastly, there is a bug in the progress bar; it is not working as intended. As I jump from Step 1 to 2, it is shown as complete. Take a look at it.

souad988 commented:
> @souad988 Most of the changes I've suggested include text changes, reducing the character limit for fields, and a few minor issues here and there. I have not come across any major issue so far, all suggestions have to do with minor improvements so that's a good thing :)
> 
> First, about helper text and placeholders. I think we don't need both. Let's remove helper text and only keep the placeholder. Maybe there is one place where we might need to retain helper text (likely materials). Use the text I've suggested for various fields. Once we finalize the changes, I can suggest some changes for the translation file for "Hindi."
> 
> Then, I saw that there is quite a lot of stuff that gets printed on the console. Make sure there are no errors, and they don't get printed as well.
> 
> Lastly, there is a bug in the progress bar; it is not working as intended. As I jump from Step 1 to 2, it is shown as complete. Take a look at it.

Hi @srish thanks for the review, concerning the progress bar i think it shows as complete when you change one field and click next  only on edit mode because the form is already valid so the user should be able to save

srish reviewed (changes requested):
@souad988 I also noticed that if we add a video url (e.g., youtube) thumbnail appears as broken. I think this one you can file as an issue and no need to fix yourself.

srish reviewed (changes requested):
I found a bug in the "Add a video" field. If let say I enter in the video url box and exit without typing anything it assumes an input and screen freezes thereafter. On the console, I see the following msg:

> createActivityScripts.js:171 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'type')
    at Object.<anonymous> (createActivityScripts.js:171:1)
    at runTest (createValidation.js:14:1)
    at validate (createValidation.js:101:1)
    at mixed.js:292:1
    at Array.map (<anonymous>)
    at mixed.js:291:1
    at async Promise.all (:3000/index 0)
(anonymous) @ createActivityScripts.js:171
runTest @ createValidation.js:14
validate @ createValidation.js:101
(anonymous) @ mixed.js:292
(anonymous) @ mixed.js:291
Promise.then (async)
handleInputBlur @ inputScripts.js:4
onBlur @ input.jsx:71
handleBlur @ InputBase.js:332
callCallback @ react-dom.development.js:3945
invokeGuardedCallbackDev @ react-dom.development.js:3994
invokeGuardedCallback @ react-dom.development.js:4056
invokeGuardedCallbackAndCatchFirstError @ react-dom.development.js:4070
executeDispatch @ react-dom.development.js:8243
processDispatchQueueItemsInOrder @ react-dom.development.js:8275
processDispatchQueue @ react-dom.development.js:8288
dispatchEventsForPlugins @ react-dom.development.js:8299
(anonymous) @ react-dom.development.js:8508
batchedEventUpdates$1 @ react-dom.development.js:22396
batchedEventUpdates @ react-dom.development.js:3745
dispatchEventForPluginEventSystem @ react-dom.development.js:8507
attemptToDispatchEvent @ react-dom.development.js:6005
dispatchEvent @ react-dom.development.js:5924
unstable_runWithPriority @ scheduler.development.js:468
runWithPriority$1 @ react-dom.development.js:11276
discreteUpdates$1 @ react-dom.development.js:22413
discreteUpdates @ react-dom.development.js:3756
dispatchDiscreteEvent @ react-dom.development.js:5889
VM124:2 Uncaught ReferenceError: process is not defined
    at 4043 (<anonymous>:2:13168)
    at r (<anonymous>:2:306599)
    at 8048 (<anonymous>:2:9496)
    at r (<anonymous>:2:306599)
    at 8641 (<anonymous>:2:1379)
    at r (<anonymous>:2:306599)
    at <anonymous>:2:315627
    at <anonymous>:2:324225
    at <anonymous>:2:324229
    at e.onload (index.js:1:1)

srish reviewed (changes requested):
There is another bug around creation steps. If I enter only the description for a step but not the image, I immediately see no error, and the "Submit for review" button also gets enabled and the progress bar appears as complete. When I click on it, I start seeing the submission in progress, and there is a loader on the screen, but the activity never gets published or created, it seems.

srish commented:
> Hi @srish thanks for the review, concerning the progress bar i think it shows as complete when you change one field and click next only on edit mode because the form is already valid so the user should be able to save

Yeah, I checked the progress bar again now that my backend setup is working fully, I think the behavior seems right for now.

souad988 committed:
Fix video url preview and empty url issues

souad988 commented:
> There is another bug around creation steps. If I enter only the description for a step but not the image, I immediately see no error, and the "Submit for review" button also gets enabled and the progress bar appears as complete. When I click on it, I start seeing the submission in progress, and there is a loader on the screen, but the activity never gets published or created, it seems.
the image is optional for making steps and the educator can describe creation process in only one step ,  I think the loading issue was due to something else i tried to reproduce the same issue but didn't happen

souad988 commented:
> @souad988 I also noticed that if we add a video url (e.g., youtube) thumbnail appears as broken. I think this one you can file as an issue and no need to fix yourself.

I added a refactorUrl function to fix this issue for youtube, vimeo and google drive videos  and also added a check for empty urls , so preview image of video and video url is correctly  working.

**next todo for video field:**
  - add upload to Cloudinary. 
  - and clear video url input on delete preview.
